R/visr_plot.R

```diff
@@ -185,37 +228,39 @@ visr.survfit <- function(
   if (is.null(y_ticks) & is.character(fun)){
     y_ticks <- base::switch(
       fun,
-      surv = pretty(c(0,1), 5),
-      log =  pretty(round(c(ymin,ymax), 0), 5),
-      event = pretty(c(0,1), 5),
-      cloglog = pretty(round(c(ymin,ymax), 0), 5),
-      pct = pretty(c(0,100), 5),
-      logpct = pretty(c(0,5), 5),
-      cumhaz =  pretty(round(c(ymin,ymax), 0), 5),
+      surv = pretty(c(0, 1), 5),
+      log =  pretty(round(c(ymin, ymax), 0), 5),
+      event = pretty(c(0, 1), 5),
+      cloglog = pretty(round(c(ymin, ymax), 0), 5),
+      pct = pretty(c(0, 100), 5),
+      logpct = pretty(c(0, 5), 5),
+      cumhaz =  pretty(round(c(ymin, ymax), 0), 5),
       stop("Unrecognized fun argument")
     )
-  } else if (is.null(y_label) & is.function(fun)) {
-    stop("Error in visr: No Y label defined. No default is available when `fun` is a function.")
+  } else if (is.null(y_ticks) & is.function(fun)) {
+    
+    y_ticks = pretty(round(c(ymin, ymax), 0), 5)
+    
   }
-
+  
 # Plotit -----------------------------------------------------
 
   yscaleFUN <- function(x) sprintf("%.2f", x)
 
   gg <- ggplot2::ggplot(tidy_object, ggplot2::aes(x = time, group = strata)) +
     ggplot2::geom_step(ggplot2::aes(y = est, col = strata)) + 
-    ggplot2::scale_x_continuous(name = paste0("\n", x_label),
+    ggplot2::scale_x_continuous(name = x_label,
```

Comment:
The rendering of expression objects convinced me.
I'm still in favor of making sure that we follow our philosophy of decent graphs rather than writing plain wrappers for graphs that still don't look pleasing. Can we have another way of defaulting the placing of labels with more spacing?

Reply:
We are not yet calling `visR::apply_theme()` from within `visR::visr()` but that could cover this I think. I'll be modifying this function again today anyway to respond to the raised issues. Then I'll also add a parameter for this spacing and add a reasonable default. Then the survival plots generated by `visR::visr()` would automatically look a little bit nicer (spacing, no grid, colours, theme_bw, etc) without scattering styling over different functions. What do you think about that?

Reply:
You are amazing ! Thanks

Reply:
Okay, I'll make a separate branch for this experiment though :+1: